### PR TITLE
Implement and link ac_t_umbrella

### PR DESCRIPTION
**Pull request by Hexalotl**
Adding in ac_t_umbrella

### CODE REVIEW

File: src/ac_t_umbrella.c [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+
+static void aTUMB_calc_model_scale_sub(xyz_t* scale, UMBRELLA_ACTOR* umbrella, int idx) {
+  static f32 e_open_sect_table[6] = { 0.0f, 7.0f, 11.0f, 18.0f, 22.0f, 26.0f };
+  static f32 e_open_scale_table[12] = {
+    0.0f, 0.0f, 0.0f, 0.0f, 0.5f, 1.0f,0.5f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f
+  };
+
+  static f32 kasa_open_sect_table[4] = { 0.0f, 15.0f, 22.0f, 26.0f };
+  static f32 kasa_open_scale_table[8] = {
+    3.0f, 0.15f, 3.0f, 0.15f, 1.0f, 1.0f, 0.89999998, 1.0f
```
Comment (Prakxo) [inline src/ac_t_umbrella.c:133]: Doesn't 0.9 match here?

File: include/ac_t_umbrella.h [hunk 2]
```diff
@@ -2,14 +2,39 @@
...
+
+// Size == 0x1F4 (500)
+typedef struct t_umbrella_s {
+    TOOLS_ACTOR tools_class;
+    UMBRELLA_PROC action_proc;
+    int action;
+    f32 frame;
+    xyz_t scale_e;
+    xyz_t scale_kasa;
+    BOOL opened_fully;
```
Comment (Prakxo) [inline include/ac_t_umbrella.h:24]: We never really put them, but it's personal preference if it's shown that is a bool

**Review by Prakxo — COMMENTED** [review]
Good job! I added some remarks

File: src/ac_t_umbrella.c (hunk 1 above)
Comment (Hexalotl) [reply src/ac_t_umbrella.c]: Yup! It does match to `0.9f`

File: src/ac_t_umbrella.c [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+  f32* sect_table;
+
+  corresponding_sector = 0;
+  table = data_table[umbrella->action] + idx;
+  sector_idx = table->table_length - 1;
+  frame = (int)umbrella->frame;
+  sect_table = table->sect_table + 1;
+
+  for(sector_idx; sector_idx > 0; sector_idx--) {
+    if (frame <= (int)*sect_table) {
```
Comment (Cuyler36) [inline src/ac_t_umbrella.c:181]: I think that sect_table[0] is preferable here, as you use it later.

File: src/ac_t_umbrella.c [hunk 4]
```diff
@@ -0,0 +1,386 @@
...
+
+    sect_table++;
+    corresponding_sector++;
+  }
+
+  scale_idx_offset = corresponding_sector * 2;
+  scale_table = table->scale_table;
+  num2 = scale_table[scale_idx_offset];
+  num1 = (f32) frame / (sect_table[0] - sect_table[-1]);
+  num3 = scale_table[scale_idx_offset+ 1];
```
Comment (Cuyler36) [inline src/ac_t_umbrella.c:194]: There's a missing space on the addition operator here.

File: src/ac_t_umbrella.c [hunk 5]
```diff
@@ -0,0 +1,386 @@
...
+  num4 = scale_table[scale_idx_offset + 3];
+  num5 = scale_table[scale_idx_offset + 2];
+
+  scale->x = num2 + num1 * (num5 - num2);
+  scale->y = num3 + num1 * (num4 - num3);
```
Comment (Cuyler36) [inline src/ac_t_umbrella.c:199]: I don't like that generic numX names are used here. num2 is the base scale for X, num3 is the base scale for Y, num1 is the modifier percentage, num5 is the max scale for X and num4 is the max scale for Y.

**Pull request merged**